src/Controllers/ShareDraftController.php

```diff
@@ -79,28 +81,19 @@ public function preview(HTTPRequest $request)
 
         $latest = Versioned::get_latest_version(SiteTree::class, $shareToken->PageID);
 
-        $controller = $this->getControllerFor($page);
-
         if (!$shareToken->isExpired() && $page->generateKey($shareToken->Token) === $key) {
             Requirements::css('silverstripe/sharedraftcontent: client/dist/styles/bundle-frontend.css');
 
             // Temporarily un-secure the draft site and switch to draft
             $oldSecured = $this->getIsDraftSecured($session);
-            $oldMode = Versioned::get_reading_mode();
+            $oldMode = Versioned::get_default_reading_mode();
```

Comment:
I'm not sure why you're switching `reading_mode()` to `default_reading_mode()`.  I've had a quick look through the codebase for Director::handleRequest() though I cannot find any reference to reading_mode that would make a difference?  Can you point to some code as to why this is necessary?

Reply:
Because I updated to use `Director::handleRequest()` (this resets the `reading_mode` but not the `default_reading_mode`)

Reply:
The "Live" mode is being set because the `VersionedHTTPMiddleware::process` will apply it based on the `HTTPRequest` (which will not have the `stage=Stage` parameter) is being processed. See `Versioned::choose_site_stage()` which is called

Reply:
OK that makes sense. I've added a suggested comment below to explain this to future developers

Reply:
added your comment